shared/teams/team/settings/index.js

```diff
@@ -0,0 +1,272 @@
+    this.setState({[key]: newSetting}, this.setPublicitySettingsChanged)
+  }
+
+  setPublicitySettingsChanged = () => {
+    const publicitySettingsChanged = !(
+      this.state.newIgnoreAccessRequests === this.props.ignoreAccessRequests &&
+      this.state.newOpenTeam === this.props.openTeam &&
+      this.state.newOpenTeamRole === this.props.openTeamRole &&
+      this.state.newPublicityAnyMember === this.props.publicityAnyMember &&
+      this.state.newPublicityMember === this.props.publicityMember &&
+      this.state.newPublicityTeam === this.props.publicityTeam
+    )
+    this.setState({publicitySettingsChanged})
+  }
+
+  onSaveSettings = () => {
+    this.props.savePublicity({
+      ignoreAccessRequests: this.state.newIgnoreAccessRequests,
+      openTeam: this.state.newOpenTeam,
+      openTeamRole: this.state.newOpenTeamRole,
+      publicityAnyMember: this.state.newPublicityAnyMember,
+      publicityMember: this.state.newPublicityMember,
+      publicityTeam: this.state.newPublicityTeam,
+    })
+  }
+
+  onSetOpenTeamRole = () => {
+    this.props.setOpenTeamRole(this.state.newOpenTeamRole, this.setSettings('newOpenTeamRole'))
+  }
+
+  render() {
+    const teamsLink = 'keybase.io/popular-teams'
+    return (
+      <ScrollView
+        style={{
+          ...globalStyles.flexBoxColumn,
+          alignSelf: 'stretch',
+          flexBasis: 0,
+          flexGrow: 1,
+        }}
+        contentContainerStyle={{padding: globalMargins.medium}}
+      >
+        <Box
+          style={{
+            ...globalStyles.flexBoxRow,
+          }}
+        >
+          <Box style={{...globalStyles.flexBoxColumn, alignItems: 'center'}}>
+            <Checkbox
+              checked={this.state.newPublicityMember}
+              disabled={!this.props.yourOperations.setMemberShowcase}
+              label=""
+              onCheck={this.setSettings('newPublicityMember')}
+              style={{paddingRight: globalMargins.xtiny}}
+            />
+          </Box>
+          <Box style={{...globalStyles.flexBoxColumn, flexShrink: 1}}>
+            <Text
+              style={{
+                color: this.props.yourOperations.setMemberShowcase
+                  ? globalColors.black_75
+                  : globalColors.grey,
+              }}
+              type="Body"
+            >
+              Publish team on your own profile
+            </Text>
+            <Text type="BodySmall">
+              {this.props.yourOperations.setMemberShowcase
+                ? 'Your profile will mention this team. Team description and number of members will be public.'
+                : "Admins aren't allowing members to publish this team on their profile."}
+            </Text>
+          </Box>
+        </Box>
+
+        {(this.props.yourOperations.changeOpenTeam ||
+          this.props.yourOperations.setTeamShowcase ||
+          this.props.yourOperations.setPublicityAny) && (
+          <Box style={globalStyles.flexBoxColumn}>
```

Comment:
these should be pulled into their own components. can be in the same file but it should just be its own thing. this render function is too long